pkg/storage/interfaces.go

```diff
@@ -91,7 +92,7 @@ type Interface interface {
 	Set(ctx context.Context, key string, obj, out runtime.Object, ttl uint64) error
 
 	// Delete removes the specified key and returns the value that existed at that spot.
-	Delete(ctx context.Context, key string, out runtime.Object) error
+	Delete(ctx context.Context, key string, out runtime.Object, preconditions *api.Preconditions) error
```

Comment:
I have a mild preference for making a Precondition type here, in this package, instead of depending on the api package.